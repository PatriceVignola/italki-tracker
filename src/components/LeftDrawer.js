/**
 * @prettier
 * @flow
 */

import React from 'react';
import {
  Drawer,
  Divider,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Typography,
} from '@material-ui/core';

import PeopleIcon from '@material-ui/icons/People';

type Props = {
  open: boolean,
  onClose: () => void,
};

function LeftDrawer(props: Props) {
  return (
    <Drawer open={props.open} onClose={props.onClose}>
      <div
        tabIndex={0}
        role="button"
        onClick={props.onClose}
        onKeyDown={props.onClose}
      >
        <List style={styles.list}>
          <ListItem style={styles.title}>
            <Typography variant="title">
              Italki Tracker
            </Typography>
          </ListItem>
          <Divider />
          <ListItem button>
            <ListItemIcon>
              <PeopleIcon />
            </ListItemIcon>
            <ListItemText primary="Students" />
          </ListItem>
        </List>
      </div>
    </Drawer>
  );
}

const styles = {
  list: {
    minWidth: 300,
  },
  title: {
    marginTop: 5,
    marginBottom: 10,
  },
};

export default LeftDrawer;
