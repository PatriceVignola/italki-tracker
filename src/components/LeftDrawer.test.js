/**
 * @prettier
 * @flow
 */

import React from 'react';
import {shallow} from 'enzyme';

import LeftDrawer from './LeftDrawer';

describe('LeftDrawer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<LeftDrawer open onClose={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
