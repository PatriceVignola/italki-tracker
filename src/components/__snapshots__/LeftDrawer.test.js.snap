// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftDrawer renders correctly 1`] = `
<WithStyles(Drawer)
  onClose={[Function]}
  open={true}
>
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    <WithStyles(List)
      style={
        Object {
          "minWidth": 300,
        }
      }
    >
      <WithStyles(ListItem)
        style={
          Object {
            "marginBottom": 10,
            "marginTop": 5,
          }
        }
      >
        <WithStyles(Typography)
          variant="title"
        >
          Italki Tracker
        </WithStyles(Typography)>
      </WithStyles(ListItem)>
      <WithStyles(Divider) />
      <WithStyles(ListItem)
        button={true}
      >
        <WithStyles(ListItemIcon)>
          <pure(People) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="Students"
        />
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        button={true}
      >
        <WithStyles(ListItemIcon)>
          <pure(School) />
        </WithStyles(ListItemIcon)>
        <WithStyles(ListItemText)
          primary="Courses"
        />
      </WithStyles(ListItem)>
    </WithStyles(List)>
  </div>
</WithStyles(Drawer)>
`;
